pkg/registry/pod/rest.go

```diff
@@ -59,6 +59,14 @@ func (podStrategy) ResetBeforeCreate(obj runtime.Object) {
 	}
 }
 
+// PrepareForUpdate clears fields that are not allowed to be set by end users on update.
+func (podStrategy) PrepareForUpdate(obj, old runtime.Object) {
+	newPod := obj.(*api.Pod)
+	oldPod := old.(*api.Pod)
+	newPod.Status = oldPod.Status
+	//FIXME: can Spec.Host be updated?
```

Comment:
No, `ValidatePodUpdate` allows only container.image to be updated. That said, I think it's more clear for the request to fail at the validation than simply overwriting `Spec.Host` here.